lib/linked_list.rb

```diff
@@ -7,39 +7,65 @@ def initialize
     @head = nil
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
-  def add_first(data)
-
+  # Time complexity - O(1)
+  # Space complexity - O(n)
+  def add_first(data) 
+    @head = Node.new(data, @head)
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(1)
+  # Space complexity - O(n)
   def get_first
-
+    return @head.nil? ? nil : @head.data
   end
 
-  # Time complexity - ?
-  # Space complexity - ?
+  # Time complexity - O(n)
+  # Space complexity - O(n)
   def length
```

Comment:
This works, but why do you think the space complexity is O(n)?  You are not creating a new list or doing recursion.